### PR TITLE
 prevent support scalar in MultiselectOptionset table

### PR DESCRIPTION
**Pull request by kartikkr11**
This change basically restricts a scalar value like optionsetvalue to be used in the IN operator with a MultiselectOptionset Table, this has resulted in issues where formulas in App don't work in edit mode until they are forced to be reevaluated

### CODE REVIEW

File: src/libraries/Microsoft.PowerFx.Core/Binding/Binder.cs [hunk 1]
```diff
@@ -2717,13 +2717,13 @@ private bool CheckInArgTypes(TexlNode left, TexlNode right)
                         }
 
                         var typedName = names.Single();
-                        if (!typeLeft.CoercesTo(typedName.Type))
```
Comment (lesaltzm) [inline src/libraries/Microsoft.PowerFx.Core/Binding/Binder.cs:2720]: Why these changes? Can you leave a comment about the logic?

Comment (kartikkr11) [reply src/libraries/Microsoft.PowerFx.Core/Binding/Binder.cs:2720]: updated it and added comments.

**Review by lesaltzm — APPROVED** [review]
